src/Api/PubnubApi/EndPoint/Files/SendFileOperation.cs

```diff
@@ -90,7 +91,8 @@ public SendFileOperation File(string fileNameWithFullPath)
         {
             this.sendFileFullPath = fileNameWithFullPath;
 #if !NETSTANDARD10 && !NETSTANDARD11
-            if (System.IO.File.Exists(fileNameWithFullPath))
+            // manually set filename should take precedence
+            if (System.IO.File.Exists(fileNameWithFullPath) && string.IsNullOrEmpty(sendFileName))
```

Comment:
shouldn't there be not operator in front of string.IsNullOrEmpty?
`!string.IsNullOrEmpty(sendFileName)`

Reply:
The intention was not to overwrite the filename (with the one that comes from abosute path) if it was manually set by the user with the FileName method.